packages/mask/src/plugins/Savings/SNSAdaptor/SavingsForm.tsx

```diff
@@ -0,0 +1,165 @@
+import { TextField, Button, Typography } from '@mui/material'
+import { useState, useMemo } from 'react'
+import { EthereumTokenType, useFungibleTokenBalance, useWeb3, useAccount } from '@masknet/web3-shared-evm'
+import { useI18N } from '../../../utils'
+import { useStyles } from './SavingsFormStyles'
+import { IconURLS } from './IconURL'
+import { TabType } from '../types'
+import { SavingsProtocols } from '../protocols'
+import { isLessThan, leftShift, rightShift } from '@masknet/web3-shared-base'
+import { EthereumWalletConnectedBoundary } from '../../../web3/UI/EthereumWalletConnectedBoundary'
+import { EthereumChainBoundary } from '../../../web3/UI/EthereumChainBoundary'
+import { ActionButtonPromise } from '../../../extension/options-page/DashboardComponents/ActionButton'
+
+export interface SavingsFormProps {
+    chainId: number
+    selectedProtocol: number
+    tab: TabType
+    onClose?: () => void
+    onSwapDialogOpen?: () => void
+}
+
+export function SavingsForm({ chainId, selectedProtocol, tab, onClose, onSwapDialogOpen }: SavingsFormProps) {
+    const { t } = useI18N()
+    const { classes } = useStyles()
+    const protocol = SavingsProtocols[selectedProtocol]
+    const targetChainId = chainId
+
+    const web3 = useWeb3(false, targetChainId)
+    const account = useAccount()
+
+    const [inputAmount, setInputAmount] = useState('')
+
+    const { value: nativeTokenBalance, loading: loadingNativeTokenBalance } = useFungibleTokenBalance(
+        EthereumTokenType.Native,
+        '',
+        targetChainId,
+    )
+
+    //#region form controls
+    const formattedBalance = leftShift(Number.parseInt(nativeTokenBalance || '0', 16), 18)
```

Comment:
Here it didn't format anything.

```suggestion
    const balance = leftShift(Number.parseInt(nativeTokenBalance || '0', 16), 18)
```

Reply:
```suggestion
    const formattedBalance = leftShift(new BigNumber(nativeTokenBalance || '0', 16), 18)
```